backend/store/etcd/watchers.go

```diff
@@ -123,3 +125,52 @@ func GetResourceWatcher(ctx context.Context, client *clientv3.Client, key string
 
 	return ch
 }
+
+// GetResourceV3Watcher returns a channel that emits WatchEventResourceV3
+// structs notifying that a corev3.Resource has been created, deleted or
+// updated.
+func GetResourceV3Watcher(ctx context.Context, client *clientv3.Client, key string) <-chan store.WatchEventResourceV3 {
+	w := Watch(ctx, client, key, true)
+	c := make(chan store.WatchEventResourceV3, 1)
+
+	go func() {
+		defer close(c)
```

Comment:
Can you add a deferred logger statement here at info level, so we can see the watcher shutting down?